testutil/integration.go

```diff
@@ -0,0 +1,166 @@
+package testutil
+
+import (
+	"strconv"
+
+	sdkerrors "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	govv1beta1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
+	"github.com/evmos/ethermint/encoding"
+
+	"github.com/evmos/evmos/v9/app"
+)
+
+// SubmitProposal delivers a submit proposal tx for a given gov content.
+// Depending on the content type, the eventNum needs to specify submit_proposal
+// event.
+func SubmitProposal(
+	ctx sdk.Context,
+	appEvmos *app.Evmos,
+	pk *ethsecp256k1.PrivKey,
+	content govv1beta1.Content,
+	eventNum int,
+) (id uint64, err error) {
+	accountAddress := sdk.AccAddress(pk.PubKey().Address().Bytes())
+	stakeDenom := stakingtypes.DefaultParams().BondDenom
+
+	deposit := sdk.NewCoins(sdk.NewCoin(stakeDenom, sdk.NewInt(100000000)))
+	msg, err := govv1beta1.NewMsgSubmitProposal(content, deposit, accountAddress)
+	if err != nil {
+		return 0, err
+	}
+	res, err := DeliverTx(ctx, appEvmos, pk, msg)
+	if err != nil {
+		return 0, err
+	}
+
+	submitEvent := res.GetEvents()[eventNum]
+	if submitEvent.Type != "submit_proposal" || string(submitEvent.Attributes[0].Key) != "proposal_id" {
+		return 0, sdkerrors.Wrapf(sdkerrors.Error{}, "eventNumber %d in SubmitProposal calls %s instead of submit_proposal", eventNum, submitEvent.Type)
+	}
+
+	proposalID, err := strconv.ParseUint(string(submitEvent.Attributes[0].Value), 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return proposalID, nil
```

Comment:
Nit: remove the error check and return the `strconv.ParseUint()` as it returns the same signature as the current func
```suggestion
	return strconv.ParseUint(string(submitEvent.Attributes[0].Value), 10, 64)
```